app-web/src/pages/index.js

```diff
@@ -106,18 +106,18 @@ const getSearchResultTotal = results => {
 /**
  * returns a resource preview components
  * @param {Array} resources the list of siphon resources
- * @param {string} queryExists the search query
  * @param {Array} results the list of searched resources
+ * @param {Srting} title the title of currtion card section
  */
-const getResourcePreviews = (resources, queryExists, results = [], title) => {
+const getResourcePreviews = (resources, results = [], title) => {
   const resourcesSelector = selectResourcesGroupedByType();
   let resourcesToShow = [];
   if (!isNull(results) && results.length > 0) {
     //map the search index results to the resources. Its important to do it in this order,
     //since the index results are return in order based on relevance
     resourcesToShow = results.flatMap(result => {
-      return resources.filter(resource => result.id === resource.id);
-    });
+      return resources.filter(resource => result.id === resource.id); //this already remove duplicate, and have some potential issue as some resource will be filter out, same item has different id in resources and result.
+    }); // TODO, investigate in above two lines.
```

Comment:
??

Reply:
i think we need to do some further investigtion on how ids are intersected between results and resources.